### DISCUSSION TITLE
Missing content-encoding: gzip for vnd.npm.install-v1+json requests

### DISCUSSION THREAD

**Discussion by arcanis** [npm/feedback#749] [Registry] [answered]
Adding the `Accept: application/vnd.npm.install-v1+json` header to registry requests cause the registry to "forget" to compress the response (note the lack of `Content-Encoding` header in the response, despite the `Accept-Encoding` header being set); is that expected?

```
❯ curl -I -H "Accept-Encoding: gzip" -H 'Accept: application/vnd.npm.install-v1+json' https://registry.yarnpkg.com/typanion

HTTP/2 200
date: Fri, 12 Aug 2022 11:55:21 GMT
content-type: application/vnd.npm.install-v1+json
cf-ray: 73990581f86b999e-CDG
accept-ranges: bytes
access-control-allow-origin: *
age: 138
cache-control: public, max-age=300
etag: "985abae9fe8a0940795137f1da9ac93d"
last-modified: Sun, 22 May 2022 12:11:08 GMT
vary: accept-encoding, accept
cf-cache-status: HIT
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
x-amz-replication-status: COMPLETED
server: cloudflare


❯ curl -I -H "Accept-Encoding: gzip" https://registry.yarnpkg.com/typanion

HTTP/2 200
date: Fri, 12 Aug 2022 11:55:37 GMT
content-type: application/json
cf-ray: 739905e488b332c2-CDG
access-control-allow-origin: *
age: 174
cache-control: public, max-age=300
etag: W/"e64a2c7ff74e78dc9e0dfa8346ce4dec"
last-modified: Sun, 22 May 2022 12:11:08 GMT
vary: accept-encoding, accept
cf-cache-status: HIT
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
x-amz-replication-status: COMPLETED
server: cloudflare
content-encoding: gzip
```

**Comment by inomdzhon** [comment]
For more context:
- [[npm registry] How to force gzip compression](https://github.com/orgs/community/discussions/54419)
- https://github.com/npm/registry-issue-archive/issues/365

**Comment by broofa** [comment]
FWIW, I just opened [a similar-but-slightly-different discussion here](https://github.com/npm/feedback/discussions/1014).  (Should have commented here, but forgot to look for similar issues first.  Dooh!)  It provides additional information about the size impact compared to requests made with `Accept: application/json`.

... and for anyone at NPM who might be paying attention: This issue negates any benefit the `application/vnd.npm.install-v1+json` format might have.  NPM registry servers are sending clients 1/2 the amount of information (smaller data structures) but twice the data (uncompressed responses).

Given that there's been no response from NPM on this, I expect to switch npmgraph.js.org back to the `application/json` format shorty.  😖

**Comment by hashtagchris** [comment, marked as answer]
👋🏼  Hello from npm! Thanks for reporting the issue. We are tracking this issue but don't have a timeline for fixing it.

**Reply by hashtagchris** [reply]
Update: We have deployed a fix for the issue. Please let us know if your responses aren't gzipped.
